### PR TITLE
EOS-27087: Improve version compatibility check.

### PR DESCRIPTION
**Pull request by shriya-deshmukh**
Signed-off-by: Shriya Deshmukh <shriya.deshmukh@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: src/provisioner/provisioner.py [hunk 1]
```diff
@@ -230,10 +230,8 @@ def _provision_components(cortx_conf: ConfigStore, mini_prov_interfaces: list, a
                         component_name, err)
                 # Update version for each component if upgrade successful.
                 if apply_phase == PROVISIONING_STAGES.UPGRADE.value:
-                    component_version = CortxProvisioner.cortx_release.get_version(component_name)
+                    component_version = CortxProvisioner.cortx_release.get_component_version(component_name)
```
Comment (ujjwalpl) [inline src/provisioner/provisioner.py:233]: Fix indentation

File: src/setup/cortx_deploy [hunk 2]
```diff
@@ -50,9 +50,9 @@ function conf_get {
 function apply_config {
     for file in "$SOLUTION_CONFIG"/*
     do
-        if [ -f "$file" ]
+        [ -f "$file" ] &&
+        if [[ "$(conf_get $(yaml_config $file) 'cluster')" != "null" || "$(conf_get $(yaml_config $file) 'cortx')" != "null" ]]
```
Comment (ujjwalpl) [inline src/setup/cortx_deploy:54]: What is yaml_config ? If its name of the function then its incorrect. Function names should typically be verb e.g. parse_url if you really meant that.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: src/provisioner/provisioner.py (hunk 1 above)
Comment (shriya-deshmukh) [reply src/provisioner/provisioner.py]: Done

File: src/setup/cortx_deploy (hunk 2 above)
Comment (shriya-deshmukh) [reply src/setup/cortx_deploy]: Renamed yaml_config() to make_url()

**Review by ujjwalpl — APPROVED** [review]

**Review by tanujashinde0405 — APPROVED** [review]

**Pull request merged**